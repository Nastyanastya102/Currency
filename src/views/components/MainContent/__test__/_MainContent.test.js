import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import expect from 'expect';

import Content from '../MainContent';

const props = {
  gettingData: {
    pending: false,
    error: false,
    dataFromAPI: [],
    formGridFrome: '',
    formGridTo: '',
    inputText: 0,
    date: '',
  },
  getTextInput: jest.fn(),
  setToCur: jest.fn(),
  getSumValue: jest.fn(),
  fetchProductsPending: jest.fn(),
};
describe('Should render MainContent', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Content {...props} />);
  });

  it('renders correctly', () => {
    wrapper = shallow(<Content {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should return Loading text', () => {
    const nextProps = {
      ...props,
      gettingData: {
        ...props.gettingData,
        pending: true,
      },
    };
    wrapper = shallow(<Content {...nextProps} />);
    expect(wrapper.find('p').text()).toEqual('Loading...');
  });

  it('Should return 3 Inputs and Sum component', () => {
    expect(wrapper.find('Input')).toHaveLength(3);
    expect(wrapper.find('Connect(Sum)')).toHaveLength(1);
  });

  it('Should simulate onSubmit', () => {
    wrapper.find('Form').simulate('submit', {
      preventDefault: () => {},
    });
  });
});
