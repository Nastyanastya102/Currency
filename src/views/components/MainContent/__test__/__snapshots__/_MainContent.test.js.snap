// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MainContent renders correctly 1`] = `
<Form
  inline={false}
  onSubmit={[Function]}
>
  <FormRow>
    <Input
      as="input"
      controlId="formGridInput"
      event={[Function]}
      placeholder="Enter value"
      text="Amount"
      value={0}
    />
    <Input
      as="select"
      controlId="formGridFrome"
      event={[Function]}
      list={Array []}
      name="from"
      text="From"
      value=""
    />
    <Input
      as="select"
      controlId="formGridTo"
      event={[Function]}
      list={Array []}
      name="to"
      text="To"
      value=""
    />
  </FormRow>
  <Connect(Sum) />
</Form>
`;
